scripts/release_scripts/run_macaron.sh

```diff
@@ -598,7 +599,21 @@ else
     fi
 fi
 
+# Handle tokens.
+set +u
+echo "" > ${TOKEN_FILE}
+{
+    echo "GITHUB_TOKEN=${GITHUB_TOKEN}" >> ${TOKEN_FILE}
+    echo "MCN_GITLAB_TOKEN=${MCN_GITLAB_TOKEN}" >> ${TOKEN_FILE}
+    echo "MCN_SELF_HOSTED_GITLAB_TOKEN=${MCN_SELF_HOSTED_GITLAB_TOKEN}"
+} >> ${TOKEN_FILE}
+mount_file "macaron_env_file" ${TOKEN_FILE} ${MACARON_WORKSPACE}/${TOKEN_FILE} "rw,Z"
+set -u
+
 # Force docker to use linux/amd64 platform in order to make docker use emulation on ARM host platforms.
+# We need the `docker run` command to be within `set +e` so that when it returns a non-zero status code,
+# it doesn't exit right away and still run the token file cleaning up command.
+set +e
```

Comment:
Are you certain that nothing can fail between when the token file is written, and when exit-on-error is disabled here? Should it be disabled earlier (perhaps even before writing the token file) to be safe? It seems to me that exit-on-error is really dangerous if you have cleanup that must be done.

Reply:
Hmmm, that it's true, I agree that we should disable exit-on-error for the block that create the token file. I will look into addressing it.

Reply:
Addressed in https://github.com/oracle/macaron/pull/780/commits/2f2b2cae99fcf096d47deb66943aef8837b7cd4d

Reply:
My solution in https://github.com/oracle/macaron/commit/2f2b2cae99fcf096d47deb66943aef8837b7cd4d ensures that exit-on-error is not enable for the whole scope of the existence of the token file. This will ensure that the clean up code will always be run after the life time of the token file.
There is an alternative solution, where we uses a "trap - https://www.gnu.org/software/bash/manual/html_node/Bourne-Shell-Builtins.html#index-trap. According to the documentation of `set -e` - https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html, "trap" can be use to run any command before the shell exits early (caused by `set -e`):
```base
trap \
    # $TOKEN_FILE here should be expanded now, instead of when the signal is trapped.
    "rm -rf '$TOKEN_FILE'" \
    ERR
``` 
I am fine with both, but I think I prefer the first solution as it doesn't rely on the exit on error behavior, which is not safe (when we have to deal with the token file) as pointed out by @nicallen .